pineappl_py/pineappl/fk_table.py

```diff
@@ -50,6 +52,42 @@ def optimize(self, assumptions = "Nf6Ind"):
             assumptions = FkAssumptions(assumptions)
         return self._raw.optimize(assumptions._raw)
 
+    def convolute_with_one(
+        self,
+        pdg_id,
+        xfx,
+        bin_indices=np.array([], dtype=np.uint64),
+        lumi_mask=np.array([], dtype=bool),
+    ):
+        r"""Convolute FkTable with a pdf.
+
+        Parameters
+        ----------
+            pdg_id : int
+                PDG Monte Carlo ID of the hadronic particle `xfx` is the PDF for
+            xfx : callable
+                lhapdf like callable with arguments `pid, x, Q2` returning x*pdf for :math:`x`-grid
+            bin_indices : sequence(int)
+                A list with the indices of the corresponding bins that should be calculated. An
+                empty list means that all orders should be calculated.
+            lumi_mask : sequence(bool)
+                Mask for selecting specific luminosity channels. The value `True` means the
+                corresponding channel is included. An empty list corresponds to all channels being
+                enabled.
+
+        Returns
+        -------
+            list(float) :
+                cross sections for all bins, for each scale-variation tuple (first all bins, then
+                the scale variation)
+        """
+        return self.raw.convolute_with_one(
+            pdg_id,
+            xfx,
+            np.array(bin_indices),
+            np.array(lumi_mask),
+        )
+
```

Comment:
This is unnecessary. The `PyWrapper.__getattr__` is already exposing it, from the underlying object. The `numpy` crate is doing the NumPy-to-ndarray conversion.

Reply:
Which part is unnecessary? The `np.array`?

Reply:
Then read the docs :P

https://pyo3.rs/v0.21.2/function/signature#using-pyo3signature--

Reply:
I proposed something in c42be03, but I haven't tested yet (especially the value assigned to the defaults) because I still have to locally rebase on #245.